### PR TITLE
RDM-11213: Disregard [CREATOR] role

### PR DESCRIPTION
**Pull request by tomxelliott**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-11213

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperation.java [hunk 1]
```diff
@@ -235,7 +235,9 @@ public void removeCaseUserRoles(List<CaseAssignedUserRoleWithOrganisation> caseU
             cauRolesByCaseId.entrySet().stream()
             .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().stream()
                 // filter out no organisation_id
-                .filter(caseUserRole -> StringUtils.isNoneBlank(caseUserRole.getOrganisationId()))
+                .filter(caseUserRole ->
+                        StringUtils.isNoneBlank(caseUserRole.getOrganisationId())
+                            && !caseUserRole.getCaseRole().equals(CREATOR.getRole()))
```
Comment (MSancaktutar) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperation.java:240]: `equalsIgnoreCase` will be safer.

File: src/test/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperationTest.java [hunk 2]
```diff
@@ -369,6 +370,22 @@ void shouldIncrementOrganisationUserCountForSingleNewRelationship() {
                 .incrementSupplementaryData(CASE_REFERENCE.toString(), getOrgUserCountSupDataKey(ORGANISATION), 1L);
         }
 
+        @Test
+        @DisplayName("should not increment organisation user count for single new case-user with [CREATOR] role")
```
Comment (MSancaktutar) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperationTest.java:374]: Adding a decrement test could also be useful, couldn't it?

File: src/test/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperationTest.java [hunk 3]
```diff
@@ -414,6 +431,29 @@ void shouldIncrementOrganisationUserCountOnlyOnceForRepeatNewRelationship() {
                 .incrementSupplementaryData(CASE_REFERENCE.toString(), getOrgUserCountSupDataKey(ORGANISATION), 1L);
         }
 
+        @Test
+        @DisplayName("should increment organisation user count for new case-user relationship without [CREATOR] role")
```
Comment (MSancaktutar) [inline src/test/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperationTest.java:435]: Adding a decrement test could also be useful, couldn't it?

**Review by MSancaktutar — CHANGES_REQUESTED** [review]
Pls see my 3 comments.

**Review by MSancaktutar — APPROVED** [review]
LGTM

**Review by danlysiak — APPROVED** [review]
👍

**Comment by MSancaktutar** [comment]
Merged as confirmed by QA.

**Pull request merged**